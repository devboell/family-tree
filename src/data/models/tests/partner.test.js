import knex from '../../connectors'
import PartnerModel from '../partner'

describe('PartnerModel', () => {
  beforeEach(async () => {
    await knex.migrate.rollback()
    await knex.migrate.latest()
    await knex.seed.run()
  })

  afterEach(async () => {
    await knex.migrate.rollback()
  })

  afterAll(() => {
    knex.destroy()
  })

  it('findByPersonId', async () => {
    const partners = await PartnerModel.findByPersonId(4)
    expect(partners).toMatchSnapshot()
  })

  it('findChildren', async () => {
    const children = await PartnerModel.findChildren([2, 3, 4])
    expect(children).toMatchSnapshot()
  })

  it('unassignChildren', async () => {
    await PartnerModel.unassignChildren([2, 3, 4])
    const children = await PartnerModel.findChildren([2, 3, 4])
    expect(children).toMatchSnapshot()
  })
})
