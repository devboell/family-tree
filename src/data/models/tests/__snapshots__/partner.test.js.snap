// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerModel findByPersonId 1`] = `
Array [
  Object {
    "birthDate": 1080,
    "bornToId": null,
    "deathDate": 1118,
    "gender": "F",
    "id": 5,
    "isMonarch": 0,
    "name": "Matilde of Scotland",
    "relationId": 2,
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1103,
    "bornToId": null,
    "deathDate": 1151,
    "gender": "F",
    "id": 14,
    "isMonarch": 0,
    "name": "Adeliza of Louvain",
    "relationId": 5,
    "title": "Queen of England",
  },
]
`;

exports[`PartnerModel findChildren 1`] = `
Array [
  Object {
    "birthDate": 1103,
    "bornToId": 2,
    "deathDate": 1120,
    "gender": "M",
    "id": 6,
    "isMonarch": 0,
    "name": "William Adelin",
    "title": "Duke of Normandy",
  },
  Object {
    "birthDate": 1102,
    "bornToId": 2,
    "deathDate": 1167,
    "gender": "F",
    "id": 7,
    "isMonarch": 0,
    "name": "Empress Matilde",
    "title": "Holy Roman Empress",
  },
  Object {
    "birthDate": 1133,
    "bornToId": 3,
    "deathDate": 1189,
    "gender": "M",
    "id": 9,
    "isMonarch": 1,
    "name": "Henry II",
    "title": "King of England",
  },
  Object {
    "birthDate": 1157,
    "bornToId": 4,
    "deathDate": 1199,
    "gender": "M",
    "id": 11,
    "isMonarch": 1,
    "name": "Richard I",
    "title": "King of England",
  },
  Object {
    "birthDate": 1166,
    "bornToId": 4,
    "deathDate": 1216,
    "gender": "M",
    "id": 12,
    "isMonarch": 1,
    "name": "John I",
    "title": "King of England",
  },
]
`;

exports[`PartnerModel unassignChildren 1`] = `Array []`;
